### PR TITLE
Fix randomness in SystemIndexMigrationIT

### PR DESCRIPTION
**Pull request by pgomulka**
SystemIndexMigrationIT was behaving non deterministically when
one of the nodes have both master and data roles.
This was due to a setup in AbstractFeatureMigrationIntegTest where
internalCluster().getCurrentMasterNodeInstance is creating a master node

relates #85164

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md)?
- If submitting code, have you built your formula locally prior to submission with `gradle check`?
- If submitting code, is your pull request against master? Unless there is a good reason otherwise, we prefer pull requests against master and will backport as needed.
- If submitting code, have you checked that your submission is for an [OS and architecture that we support](https://www.elastic.co/support/matrix#show_os)?
- If you are submitting this code for a class then read our [policy](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md#contributing-as-part-of-a-class) for that.

### CODE REVIEW

File: modules/reindex/src/internalClusterTest/java/org/elasticsearch/migration/AbstractFeatureMigrationIntegTest.java [hunk 1]
```diff
@@ -124,10 +124,16 @@ public void setupTestPlugin() {
...
 
     public <T extends Plugin> T getPlugin(Class<T> type) {
+        createMasterNode();
         final PluginsService pluginsService = internalCluster().getCurrentMasterNodeInstance(PluginsService.class);
         return pluginsService.filterPlugins(type).stream().findFirst().get();
     }
 
+    protected void createMasterNode() {
+        // most tests do not care how master node is created, and can rely on internalCluster().getCurrentMasterNodeInstance
+        // but some like SystemIndexMigrationIT need master node to be master only node
```
Comment (rjernst) [inline modules/reindex/src/internalClusterTest/java/org/elasticsearch/migration/AbstractFeatureMigrationIntegTest.java:134]: Shouldn't this have a default implementation which creates a master node?

Comment (pgomulka) [reply modules/reindex/src/internalClusterTest/java/org/elasticsearch/migration/AbstractFeatureMigrationIntegTest.java:134]: I know it looks odd. Maybe this method could return a pluginsService instead?
Let me update

File: modules/reindex/src/internalClusterTest/java/org/elasticsearch/migration/SystemIndexMigrationIT.java [hunk 2]
```diff
@@ -64,13 +64,13 @@ protected boolean forbidPrivateIndexSettings() {
         return plugins;
     }
 
-    @Before
-    public void init() {
-        // master node is created in AbstractFeatureMigrationIntegTest#setupTestPlugin when trying to access plugins
+    @Override
+    protected PluginsService getPluginService() {
         internalCluster().setBootstrapMasterNodeIndex(0);
```
Comment (rjernst) [inline modules/reindex/src/internalClusterTest/java/org/elasticsearch/migration/SystemIndexMigrationIT.java:69]: It seems odd that asking for the PluginService is mutating the cluster. Why can't this master only node be started in a `@Before`, separately from trying to get the PluginService?

Comment (pgomulka) [reply modules/reindex/src/internalClusterTest/java/org/elasticsearch/migration/SystemIndexMigrationIT.java:69]: to start a master node manually it affects other tests in this package, as they also will have to set `autoManageMasterNodes` to false
a lot of methods in InternalTestCluster are mutating (starting nodes) the cluster.
 For instance internalCluster().getCurrentMasterNodeInstance(PluginsService.class) will start a master node (default roles) if non is started yet. 
Same as internalCluster().client()

**Review by rjernst — APPROVED** [review]
It still seems odd to me, but given that there are lots of other test examples following this convention of mutation, LGTM.

File: modules/reindex/src/internalClusterTest/java/org/elasticsearch/migration/SystemIndexMigrationIT.java (hunk 2 above)
Comment (pgomulka) [reply modules/reindex/src/internalClusterTest/java/org/elasticsearch/migration/SystemIndexMigrationIT.java]: I made all tests mange master node in migration package - I feel it should look better

**Comment by pgomulka** [comment]
@elasticmachine update branch

**Review by grcevski — APPROVED** [review]
LGTM!

**Review by AthenaEryma — APPROVED** [review]

**Pull request merged**